examples/interaction/tools/legend_visibility_tool.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+
+from bokeh.layouts import gridplot
+from bokeh.models import CustomAction, CustomJS
+from bokeh.plotting import figure, show
+
+x = np.linspace(0, 4*np.pi, 100)
+y = np.sin(x)
+
+TOOLS = "pan,wheel_zoom,box_zoom,reset,save,box_select"
+
+p1 = figure(title="Legend Example", tools=TOOLS)
+
+p1.scatter(x,   y, legend_label="sin(x)")
+p1.scatter(x, 2*y, legend_label="2*sin(x)", color="orange")
+p1.scatter(x, 3*y, legend_label="3*sin(x)", color="green")
+
+p1.legend.title = "Markers"
+
+p2 = figure(title="Another Legend Example", tools=TOOLS)
+
+p2.scatter(x, y, legend_label="sin(x)")
+p2.line(x, y, legend_label="sin(x)")
+
+p2.line(x, 2*y, legend_label="2*sin(x)",
+        line_dash=(4, 4), line_color="orange", line_width=2)
+
+p2.scatter(x, 3*y, legend_label="3*sin(x)",
+           marker="square", fill_color=None, line_color="green")
+p2.line(x, 3*y, legend_label="3*sin(x)", line_color="green")
+
+p2.legend.title = "Lines"
+
+toggle_legend = CustomAction(
+    icon=".bk-tool-icon-list",
+    description="Toggle legend",
+    callback=CustomJS(
+        args=dict(legends=p1.legend + p2.legend),
+        code="""
+        export default ({legends}) => {
+            for (const legend of legends) {
+                legend.visible = !legend.visible
+            }
+        }
+        """,
+    ),
+    active_callback=CustomJS(
```

Comment:
This is cool functionality. But could there be a way to toggle it on/off each time it is clicked without defining an `active_callback` or updating an existing `callback`?

Maybe if the value is a boolean, it will set the state or have a new argument that does this. 

```suggestion
    active_callback=True
```

```suggestion
    active: True | False | None = True
```

Reply:
I added `active_callback="auto"`, which effectively implies `tool.active = !tool.active`, and exposed `active` (and `disabled`) property in `CustomAction`.  This allows to set a boolean initial state and to change state on click. Setting `active_callback=None` (the default) means that the tool isn't stateful. I may still consider adding `active=None` and removing `"auto"`, but that may conflict with the underlying definition of `active` in `Tool` model.

Reply:
I suggest adding a small comment about this in the example.

Reply:
I added alternatives to examples.